free-4.md

```diff
@@ -0,0 +1,82 @@
+
+Error messages often seem like an entirely new language to new programmers, they seem impossibly to understand and are therefore useless to them, if I created a DSL to address this specific need, it would act as an interpreter for the person, translating the language of error messages into human readable English. The domain is limited to debugging and error messages so it does not require a GPL but the complexity of the input error messages and interpretation needed require an entire language to solve the problem, so a DSL is the appropriate tool.
+
+### Why you?
+_What excites you about this idea? How did you come up with it?_
+
+In CS105, we are currently learning how to debug by looking at the assembly code and at the information we can glean from gdb, but it is an often frusterateing process as much of the error messages seem unnecessarily vague and complicated. I have been coding for the past 4-5 years and if I am still struggling with it, surely there must be something that would make this better and more accessible to new programmers.
+
+### Domain
+_Describe the project's domain in five words._
+
+Compiler error message translation to English.
+
+
+### Interface (syntax)
+_How might the user interact with the language? What does programming look 
+like? Why is this the right way to interact with the problem domain?_ 
+
+If the user gets an error message they don't understand they should be able to run a command like "interpret" which will attempt to translate the error into clearer English. Or "suggestions" which would output probably causes for the error. This would allow users to use this tool alongside the languages they are learning as a helper for when they get stuck and can't figure out what is wrong with their code. This will decrease overhead, because my language will only be used when it is actually needed and the commands will be straight forward and easy to remember.
+
+### Operation (semantics)
+_What might happen when a program runs? How does a program interact with the
+user? What kinds of errors might occur, and how might they be communicated to
+the user?_
+
+When the program in my language runs, it will read in the error message from the users program, pick out the relevant information for the user and print it out in full sentences for the user alonge with suggestions for what might be causing the error. An error could occur of my language does not recognize the error message output by the program, but in general my language would do all its work in the error domain.
+
+### Expressiveness
+_What should be easy to do in this language? What should be possible, but
+difficult? What should be impossible or very difficult?_
+
+It should be easy to get more information about the error, where it occured, likely reasons for its occurance, and clear, full sentences in English explaining the problem like a human tutor might. It should be impossibly to write programs in it that don't involve error checking.
+
```

Comment:
Is there such a thing as a program that does not involve error checking? I guess I don't follow your last point.